test/bmc/bmc_api_test.rb

```diff
@@ -21,6 +21,48 @@ def setup
     authorize user, pass
   end
 
+  def test_api_throws_401_error_when_auth_is_not_provided
+    Proxy::BMC::IPMI.stubs(:providers_installed).returns(['freeipmi'])
+    Proxy::BMC::Plugin.settings.stubs(:bmc_default_provider).returns('freeipmi')
+    auth = Object.new
+    auth.stubs(:provided?).returns(false)
+    auth.stubs(:basic?).returns(true)
+    auth.stubs(:credentials).returns(nil)
+    Proxy::BMC::Api.any_instance.stubs(:auth).returns(auth)
+    test_args = { :bmc_provider => 'freeipmi' }
+    get "/#{host}/lan/gateway", test_args
+    assert_equal 'unauthorized', last_response.body
+    assert_equal 401, last_response.status
+  end
+
+  def test_api_does_not_throw_401_error_when_auth_is_provided_and_in_basic_format
+    Proxy::BMC::IPMI.stubs(:providers_installed).returns(['freeipmi'])
+    Proxy::BMC::Plugin.settings.stubs(:bmc_default_provider).returns('freeipmi')
+    Proxy::BMC::IPMI.any_instance.stubs(:gateway).returns("192.168.1.1")
+    auth = Object.new
+    auth.stubs(:provided?).returns(true)
+    auth.stubs(:basic?).returns(true)
+    auth.stubs(:credentials).returns('username','password')
+    Proxy::BMC::Api.any_instance.stubs(:auth).returns(auth)
+    test_args = { :bmc_provider => 'freeipmi' }
+    get "/#{host}/lan/gateway", test_args
+    assert_equal 200, last_response.status
+  end
+
+  def test_api_throws_401_error_when_auth_is_not_basic
+    Proxy::BMC::IPMI.stubs(:providers_installed).returns(['freeipmi'])
+    Proxy::BMC::Plugin.settings.stubs(:bmc_default_provider).returns('freeipmi')
+    auth = Object.new
+    auth.stubs(:provided?).returns(true)
+    auth.stubs(:basic?).returns(false)
+    auth.stubs(:credentials).returns('name', 'password')
```

Comment:
Again, you only need to stub/mock auth.provided and auth.basic? methods. I'd use a mock for the latter.